### ISSUE TITLE
Can't build on OSX

### ISSUE THREAD

**Issue by AustinFelipe** [atom/atom-shell#1285]
When I try to use ./script/bootstrap.py -v, I catch this error:

    /Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python /Users/austinfelipesantos/Documents/HotProductFinder/atom-shell-0.21.3/vendor/brightray/script/bootstrap --commit 103778aa0ec3772f88e915ca9efdb941afdc85cf http://gh-contractor-zcbenz.s3.amazonaws.com/libchromiumcontent
        /Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python: can't open file '/Users/austinfelipesantos/Documents/HotProductFinder/atom-shell-0.21.3/vendor/brightray/script/bootstrap': [Errno 2] No such file or directory
        None
        Traceback (most recent call last):
          File "./script/bootstrap.py", line 127, in <module>
            sys.exit(main())
          File "./script/bootstrap.py", line 28, in main
            bootstrap_brightray(args.url)
          File "./script/bootstrap.py", line 60, in bootstrap_brightray
            LIBCHROMIUMCONTENT_COMMIT, url])
          File "/Users/austinfelipesantos/Documents/HotProductFinder/atom-shell-0.21.3/script/lib/util.py", line 152, in execute_stdout
            raise e
        subprocess.CalledProcessError: Command '['/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python', '/Users/austinfelipesantos/Documents/HotProductFinder/atom-shell-0.21.3/vendor/brightray/script/bootstrap', '--commit', '103778aa0ec3772f88e915ca9efdb941afdc85cf', 'http://gh-contractor-zcbenz.s3.amazonaws.com/libchromiumcontent']' returned non-zero exit status 2

I tried to remove the node_module folder, but the error persists.

Any idea how to solve it?

Thanks!

**Comment by AustinFelipe** [comment]
I removed everything and I followed all the steps here (https://github.com/atom/atom-shell/blob/master/docs/development/build-instructions-mac.md) again. 

Now the error is:
    ....................................................
      Extracting...
      /Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python /Users/austinfelipesantos/Documents/HotProductFinder/atom-shell/script/update.py
      Download Mantle       77060  [100.0%] 
      Download ReactiveCocoa      368427  [100.0%] 
      Download Squirrel      489118  [100.0%] 
      2015-03-23 19:51:51.012 xcodebuild[50053:836867] [MT] PluginLoading: Required plug-in compatibility UUID C4A681B0-4A26-480E-93EC-1218098B9AA0 for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin' not present in DVTPlugInCompatibilityUUIDs
      2015-03-23 19:51:51.015 xcodebuild[50054:836868] [MT] PluginLoading: Required plug-in compatibility UUID C4A681B0-4A26-480E-93EC-1218098B9AA0 for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin' not present in DVTPlugInCompatibilityUUIDs
      2015-03-23 19:51:51.015 xcodebuild[50052:836866] [MT] PluginLoading: Required plug-in compatibility UUID C4A681B0-4A26-480E-93EC-1218098B9AA0 for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin' not present in DVTPlugInCompatibilityUUIDs
      2015-03-23 19:51:51.017 xcodebuild[50053:836867] Failed to load plugin at: /Users/austinfelipesantos/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin, skipping.  Reason for failure: *** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[0]
      2015-03-23 19:51:51.017 xcodebuild[50054:836868] Failed to load plugin at: /Users/austinfelipesantos/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin, skipping.  Reason for failure: *** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[0]
      2015-03-23 19:51:51.017 xcodebuild[50052:836866] Failed to load plugin at: /Users/austinfelipesantos/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin, skipping.  Reason for failure: *** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[0]
      2015-03-23 19:51:51.860 xcodebuild[50059:836909] [MT] PluginLoading: Required plug-in compatibility UUID C4A681B0-4A26-480E-93EC-1218098B9AA0 for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin' not present in DVTPlugInCompatibilityUUIDs
      2015-03-23 19:51:51.861 xcodebuild[50059:836909] Failed to load plugin at: /Users/austinfelipesantos/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin, skipping.  Reason for failure: *** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[0]
      2015-03-23 19:51:51.869 xcodebuild[50060:836915] [MT] PluginLoading: Required plug-in compatibility UUID C4A681B0-4A26-480E-93EC-1218098B9AA0 for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin' not present in DVTPlugInCompatibilityUUIDs
      2015-03-23 19:51:51.870 xcodebuild[50060:836915] Failed to load plugin at: /Users/austinfelipesantos/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin, skipping.  Reason for failure: *** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[0]
      2015-03-23 19:51:51.882 xcodebuild[50058:836908] [MT] PluginLoading: Required plug-in compatibility UUID C4A681B0-4A26-480E-93EC-1218098B9AA0 for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin' not present in DVTPlugInCompatibilityUUIDs
      2015-03-23 19:51:51.883 xcodebuild[50058:836908] Failed to load plugin at: /Users/austinfelipesantos/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin, skipping.  Reason for failure: *** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[0]
      /Users/austinfelipesantos/Documents/HotProductFinder/atom-shell/node_modules/.bin/apm install
      dyld: lazy symbol binding failed: Symbol not found: _node_module_register
        Referenced from: /Users/austinfelipesantos/Documents/HotProductFinder/atom-shell/node_modules/atom-package-manager/node_modules/keytar/build/Release/keytar.node
        Expected in: dynamic lookup

      dyld: Symbol not found: _node_module_register
        Referenced from: /Users/austinfelipesantos/Documents/HotProductFinder/atom-shell/node_modules/atom-package-manager/node_modules/keytar/build/Release/keytar.node
        Expected in: dynamic lookup

      /Users/austinfelipesantos/Documents/HotProductFinder/atom-shell/node_modules/.bin/apm: line 40: 50071 Trace/BPT trap: 5       "$binDir/$nodeBin" --harmony_collections "$binDir/../lib/cli.js" "$@"
      None
      Traceback (most recent call last):
        File "./script/bootstrap.py", line 131, in <module>
          sys.exit(main())
        File "./script/bootstrap.py", line 36, in main
          update_atom_modules('spec')
        File "./script/bootstrap.py", line 89, in update_atom_modules
          execute_stdout([apm, 'install'])
        File "/Users/austinfelipesantos/Documents/HotProductFinder/atom-shell/script/lib/util.py", line 152, in execute_stdout
          raise e
      subprocess.CalledProcessError: Command '['/Users/austinfelipesantos/Documents/HotProductFinder/atom-shell/node_modules/.bin/apm', 'install']' returned non-zero exit status 133

**Comment by zcbenz** [comment]
Were you using Node 0.12.x? Currently Node 0.10.x is required for building.

**Comment by AustinFelipe** [comment]
Oh, yes, I was using 0.10... Ok, thank you!

**Comment by arthurakay** [comment]
Do we still need to run 0.10.x for the OSX build? 

I tried running with 0.12.x and hit a similar error as @AustinFelipe -- then I tried switching to 0.10.x (using `nvm`) but I still hit errors (though the error is now very different). I may simply open a new ticket as I'd hate to resurrect an old thread with new error messages.

I'll point out that the [docs](https://github.com/atom/electron/blob/master/docs/development/build-instructions-mac.md) say nothing about needing a specific version of node.
